### PR TITLE
[CN Number] Support extraction for “五” of “这不是五这是二百零一” and 五" of "袜子五双九点九"

### LINKED ISSUE #1745: [CN Number] Missing extractions in "袜子五双九点九", “这不是五这是二百零一”
**Describe the bug**
Missing extraction for "五双" of "袜子五双九点九" 
Missing extraction for “五” of “这不是五这是二百零一”

### PR DESCRIPTION
**Pull request by FeifanLiu98**
Fixes #1745

### CODE REVIEW

File: Specs/Number/Chinese/NumberModel.json [hunk 1]
```diff
@@ -4430,5 +4430,51 @@
         }
       }
     ]
+  },
+  {
+    "Input": "这不是五这是二百零一",
```
Comment (wgx998877) [inline Specs/Number/Chinese/NumberModel.json]: have you try "这不是四川这是五台山"，“这不是四川这是五”，“这不是一这是九台”, could you also add those three cases?

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs [hunk 2]
```diff
@@ -175,7 +175,7 @@ public static class NumbersDefinitions
...
...|米|厘|毫|[条條]|船|[车車]|[辆輛]|群|[页頁]|杯|人|[张張]|次|位|份|批|[届屆]|[级級]|[种種]|套|[笔筆]|根|[块塊]|件|座|步|[颗顆]|棵|[节節]|支|只|名|年|月|日|[号號]|朵|克|[吨噸]|磅|[码碼]|英尺|英寸|升|加[仑侖]|立方|[台臺]|套|[罗羅]|令|卷|[头頭]|箱|包|桶|袋|[块塊]|家|行|期|[层層]|度|面|所|架|把|片|[阵陣]|[间間]|等|[叠疊]|碟|下|起|手|季|部|人|小[时時]|[时時]|秒|[样樣]|章|段|星|州|款|代|维|重|[户戸]|楼|路|篇|句|键|本|生|者|字|郎|道|边|场|口|线|世|岸|金|类|番|组|卦|眼|系|声|更|带|色|战|成|轮|食|首|幡|站|股|井|流|开|刻|洲|回|宮|集|练|週|和|环|甲|处|省|里|海|遍|品|体|王|尾|新|隻|版|阶|板|侧|波|身|则|扫|房|彩|木|军|居|晚|岛|课|式|通|相|区|文|端|味|田|心|胎|班|出|连|单|事|丝|副|岁|旁|幕|些|枚|招|卡|幅|言|街|指|辈|室|堆|作|封|厢|声|城|族|圈|脸|目|排|模|夕|网|市|向|极|驱|科|提|核|村|审|刀|册|例|关|粒|局|山|寸|碗|瞬|联|游|脚|宅|线|格|入|趟|貫|界|社|肢|技|滴|问|笑|院|堂|尺|寨|档|举|盘|门|客|餐|艘|毛|丈|剑|曲|任|叶|团|派|嘴|桥|抹|枝|贯|伏|拳|列|机|盒|队|进制|栋|席|斤|词|击|题|型|宗|柱|钱|拍|剧|旬|命|扇|匹|湖|壶|觉|叉|校|泉|具|串|射|证|大批|球|横|竖|尊|轴|观|审|石|束|弹|株|领|委|栏|炮|鼎|町|帆|斗|缕|桌|针|帧|转|落|足|梯|县|投|试|帮|掌|箭|盏|锅|计|大片|学期|截|顶|屋|介|剑|桂|旗|巷|挥|晃|员|翼|池|围|勺|宿|库|棒|冠|树|缸|伙|签|揽|坨|匙|桩|顿|纸|隅|诺|案|刊|厂|杆|袭|仓|床|担|帖|屏|盏|腔|贴|窍|洞|円|坪|泡|园|馆|湾|拨|枪|职|亭|背|維|[護护戸]|樓|鍵|邊|場|線|類|組|聲|帶|戰|輪|開|練|環|處|裏|體|隻|階|側|則|掃|軍|居|島|課|式|區|連|單|絲|歲|廂|聲|臉|網|極|驅|審|冊|關|聯|遊|腳|線|貫|問|檔|舉|盤|門|劍|曲|任|葉|團|派|嘴|橋|抹|枝|貫|伏|拳|列|機|盒|隊|進制|棟|詞|擊|題|錢|壺|覺|證|大批|球|橫|豎|尊|軸|觀|審|彈|領|委|欄|釘|鬥|縷|針|幀|轉|縣|試|幫|盞|鍋|計|學期|截|頂|介|劍|桂|旗|巷|揮|晃|員|圍|勺|宿|庫|棒|冠|樹|缸|夥|簽|攬|樁|頓|紙|隅|諾|廠|桿|襲|倉|擔|盞|貼|竅|洞|坪|泡|員|館|灣|撥|槍|職|\s|$)";
```
Comment (wgx998877) [inline .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: I can not see the diff, could you point out the difference?

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: Also, please rename WhiteListRegex to AllowListRegex.

Comment (FeifanLiu98) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: I changed "是" to "[这這那]?是". The existence of "这是" or "那是" is the siganl of new sentence.

Comment (wgx998877) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: but the [这那]? is optional, it means if without that, the “是” could be extracted either? So why need this changes?

Comment (FeifanLiu98) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: Without this change, only number followed by "是" would be recognized. And number followed by "这是" or "那是" would be recognized after this change. So the "五" can be extracted.

Comment (wgx998877) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/NumbersDefinitions.cs]: If that is the case, also add "也", "不", "还" ..

File: Specs/Number/Chinese/NumberModel.json [hunk 3]
```diff
@@ -4430,5 +4430,102 @@
...
+    "Results": [
```
Comment (wgx998877) [inline Specs/Number/Chinese/NumberModel.json]: add more cases for the new comments

**Pull request merged**